### PR TITLE
fix(Modal): Give Modal programmatic focus instead of close button

### PR DESCRIPTION
**Pull request by LinKCoding**
### Overview
Update Modal to give programmatic focus to the Modal instead of the close button. 

Update Modal to give programmatic focus to the Modal instead of the close button. 

### PR Checklist

- [ ] Related to designs: 
- [ ] Related to JIRA ticket: [GM-753](https://skillsoftdev.atlassian.net/browse/GM-753)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Go to Gamut's Modal page and "Trigger Modal" (?path=/docs/molecules-modals-modal--trigger-modal) 
2. Turn on VO 
3. Click on "Open Modal" Or "Open multi-view Modal" 
4. Notice that the programmatic focus is on the Modal container (and not the close button) 
5. Tab off the Modal container (into the close button) 
6. Notice that you cannot tab back into the Modal container 
7. Go to Dialog and "Trigger Dialog" (?path=/docs/molecules-modals-dialog--trigger-dialog#triggering-a-dialog) 
8. Click on "Open Dialog"
9. repeat steps 4-6
— 
Mono
1. Use the LE Preview link in the PR!
2. Go to a ex where you can gain progress (e.g. https://tayra.codecademy.com/courses/introduction-to-javascript/lessons/variables/exercises/var?PR_ENV=le-pr-6984)
3. Turn on VO
4. Click "Reset Workspace" at the bottom of the page
5. Notice that the Modal's focus is now on the container 
6. Tab off the Modal container (into the close button) 
7. Notice that you cannot tab back into the Modal container 
...
8. Profit!

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38034)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6984)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

[GM-753]: https://codecademy.atlassian.net/browse/GM-753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.23.1-alpha.41d384.0<br/>@codecademy/gamut-kit@0.6.420-alpha.41d384.0<br/>@codecademy/styleguide@66.27.1-alpha.41d384.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://667c31c176cc4b6939b9940b--gamut-preview.netlify.app>https://667c31c176cc4b6939b9940b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/667c31c176cc4b6939b9940b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.23.1-alpha.94e463.0<br/>@codecademy/gamut-kit@0.6.420-alpha.94e463.0<br/>@codecademy/styleguide@66.27.1-alpha.94e463.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668d40670dd7fa6d19ada602--gamut-preview.netlify.app>https://668d40670dd7fa6d19ada602--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668d40670dd7fa6d19ada602>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.23.1-alpha.e06527.0<br/>@codecademy/gamut-kit@0.6.420-alpha.e06527.0<br/>@codecademy/styleguide@66.27.1-alpha.e06527.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://668d5068bfe56b891eb5054c--gamut-preview.netlify.app>https://668d5068bfe56b891eb5054c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/668d5068bfe56b891eb5054c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.5a10df.0<br/>@codecademy/gamut-kit@0.6.425-alpha.5a10df.0<br/>@codecademy/styleguide@66.29.6-alpha.5a10df.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669140650f9f5a252501ef0b--gamut-preview.netlify.app>https://669140650f9f5a252501ef0b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669140650f9f5a252501ef0b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.17cfae.0<br/>@codecademy/gamut-kit@0.6.425-alpha.17cfae.0<br/>@codecademy/styleguide@66.29.6-alpha.17cfae.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669181e7b61ecd0a4ff3c1cb--gamut-preview.netlify.app>https://669181e7b61ecd0a4ff3c1cb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669181e7b61ecd0a4ff3c1cb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.a48cd1.0<br/>@codecademy/gamut-kit@0.6.425-alpha.a48cd1.0<br/>@codecademy/styleguide@66.29.6-alpha.a48cd1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669192ea9c1981624c6b3e75--gamut-preview.netlify.app>https://669192ea9c1981624c6b3e75--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669192ea9c1981624c6b3e75>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.5def8d.0<br/>@codecademy/gamut-kit@0.6.425-alpha.5def8d.0<br/>@codecademy/styleguide@66.29.6-alpha.5def8d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66919c81951df54f991e0377--gamut-preview.netlify.app>https://66919c81951df54f991e0377--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66919c81951df54f991e0377>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.008789.0<br/>@codecademy/gamut-kit@0.6.425-alpha.008789.0<br/>@codecademy/styleguide@66.29.6-alpha.008789.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6693401a2ae3bdcf0c3aa1f2--gamut-preview.netlify.app>https://6693401a2ae3bdcf0c3aa1f2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6693401a2ae3bdcf0c3aa1f2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.1-alpha.cea96c.0<br/>@codecademy/gamut-kit@0.6.426-alpha.cea96c.0<br/>@codecademy/styleguide@67.0.1-alpha.cea96c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669abb2cf8996d92aec649b1--gamut-preview.netlify.app>https://669abb2cf8996d92aec649b1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669abb2cf8996d92aec649b1>Deploy Logs</a></p>

File: packages/gamut/src/Modals/Modal.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export const Modal: React.FC<ModalProps> = ({
             variant="primary"
             onClick={(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
               setCurrentView(currentView + 1);
-              buttonRef.current?.focus();
+              // buttonRef.current?.focus();
```
Comment (dreamwasp) [inline packages/gamut/src/Modals/Modal.tsx:153]: dead code?

File: packages/styleguide/stories/Molecules/Modals/Dialog.examples.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { useState } from 'react';
+
+export const ClickOpenDialog = () => {
+  const [open, setOpen] = useState(false);
+  const toggleOpen = () => setOpen(true);
+
+  return (
+    <>
+      <Box>
+        <FillButton onClick={toggleOpen}>Open Dialog</FillButton>
```
Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/Modals/Dialog.examples.tsx:11]: weirdly i'm unable to trigger this on Enter button press 🤔

**Review by dreamwasp — APPROVED** [review]
couple tiny things but looks great + works great

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.1-alpha.0259bc.0<br/>@codecademy/gamut-kit@0.6.426-alpha.0259bc.0<br/>@codecademy/styleguide@67.0.1-alpha.0259bc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669eadc2cc801e2d19fcd688--gamut-preview.netlify.app>https://669eadc2cc801e2d19fcd688--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669eadc2cc801e2d19fcd688>Deploy Logs</a></p>

File: packages/styleguide/stories/Molecules/Modals/Dialog.examples.tsx (hunk 2 above)
Comment (LinKCoding) [reply packages/styleguide/stories/Molecules/Modals/Dialog.examples.tsx:11]: (For documentation) Turns out this is something with Storybook and having multiple dialogs opened. 
If all the other stories on the dialog page are deleted, then the example works 🤷 

That said, it works in prod, so writing this off as a quirk of SB

**Pull request merged**